sdk/src/main/scala/com/horizen/account/api/rpc/service/EthService.scala

```diff
@@ -633,25 +618,10 @@ class EthService(
         blockContext.setEvmResult(EvmResult.emptyEvmResult())
         tagStateView.applyTransaction(requestedTx, previousTransactions.length, gasPool, blockContext)
 
-        if (config != null && config.getTracer != null) {
-          // callTracer
-          if (config.getTracer == NativeTracers.CALL_TRACER.toString) {
-            if (blockContext.getEvmResult.callTracerLogs != null) {
-              new CallTracerTransactionView(blockContext.getEvmResult)
-            } else Unit
-          }
-          // 4byteTracer
-          else if (config.getTracer == NativeTracers.FOUR_BYTE_TRACER.toString) {
-            if (blockContext.getEvmResult.fourByteTracerLogs != null) {
-              blockContext.getEvmResult.fourByteTracerLogs
-            } else Unit
-          }
-        }
-        // Struct/opcode logger
-        else {
-          if(blockContext.getEvmResult.traceLogs != null)
-            new DebugTraceTransactionView(blockContext.getEvmResult)
-        }
+        // return the tracer result from the evm
+        if(blockContext.getEvmResult.tracerResult != null) {
+          blockContext.getEvmResult.tracerResult
```

Comment:
Agree, this check was needed before because we initialized the `EvmResult` with its custom constructor, but this constructor is now removed as reported in the comment below.
Now I just check the presence of the `EvmResult`, in this way if the transaction is a simple _EOAtoEAO_ that doesn't involve the evm I'm able to return an empty response instead of `null`